### PR TITLE
Upgrade Android native build tools and dependencies

### PR DESCRIPTION
**Pull request by Doko-Demo-Doa**
This PR updates several Android build tools, Gradle wrapper and make it compatible with React Native from 0.47+

### CODE REVIEW

**Review by MujtabaFR — APPROVED** [review]

File: android/build.gradle [hunk 1]
```diff
@@ -26,15 +26,13 @@ allprojects {
 }
 
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
-
-
 android {
-  compileSdkVersion 23
-  buildToolsVersion "23.0.3"
+  compileSdkVersion 27
+  buildToolsVersion "27.1.1"
```
Comment (MujtabaFR) [inline android/build.gradle]: `27.1.1` is not valid check [this page](https://stackoverflow.com/questions/49921807/failed-to-find-build-tools-revision-27-1-1-when-i-try-to-download-says-it-has) .. must be 
`buildToolsVersion '27.0.3'`

File: android/build.gradle [hunk 2]
```diff
@@ -7,7 +7,7 @@ buildscript {
         jcenter()
```
Comment (MujtabaFR) [inline android/build.gradle]: must add `google()`

Comment (MujtabaFR) [inline android/build.gradle:7]: must also add `google()`

**Comment by Doko-Demo-Doa** [comment]
Done, I mistook `Support Version` with `Build Tools Version`.

**Comment by maxbeech** [comment]
This is a critical fix - really needs to be merged.

**Comment by martsie** [comment]
Could you test the master on your project and if it's all good then I'll bump the version and publish to npm.

**Pull request merged**